src/hw-monitor.cpp

```diff
@@ -223,4 +225,115 @@ namespace librealsense
         librealsense::copy(&value, gvd.data() + offset, 1);
         return value;
     }
+
+    int hw_monitor_extended_buffers::get_msg_length(command cmd) const
+    {
+        int msg_length = HW_MONITOR_BUFFER_SIZE;
+        if (cmd.cmd == ds::fw_cmd::GET_HKR_CONFIG_TABLE || cmd.cmd == ds::fw_cmd::SET_HKR_CONFIG_TABLE)
+        {
+            switch (cmd.param2)
+            {
+            case static_cast<int>(ds::ds6_calibration_table_id::coefficients_table_id):
+            {
+                msg_length = sizeof(ds::ds6_coefficients_table);
+                break;
+            }
+            case static_cast<int>(ds::ds6_calibration_table_id::rgb_calibration_id):
+            {
+                msg_length = sizeof(ds::ds6_rgb_calibration_table);
+                break;
+            }
+            default:
+                throw std::runtime_error(rsutils::string::from() << "command GET_HKR_CONFIG_TABLE with wromg param2");
+            }
+        }
+        return msg_length;
+    }
+
+    int hw_monitor_extended_buffers::get_number_of_chunks(int msg_length) const
+    {
+        float ratio = static_cast<float>(msg_length) / 1024;
+        int ratio_int = static_cast<int>(ratio);
```

Comment:
Sorry I was interested and had to find a simple way for doing this :)

```
int get_number_of_chunks( int length )
{
    return static_cast<int>(std::ceil( length / 1024.0f)) ;
}
```

```
    std::cout << get_number_of_chunks( 1 ) << std::endl;
    std::cout << get_number_of_chunks( 1000 ) << std::endl;
    std::cout << get_number_of_chunks( 1024 ) << std::endl;
    std::cout << get_number_of_chunks( 1025 ) << std::endl;
    std::cout << get_number_of_chunks( 64 * 1024 ) << std::endl;
    std::cout << get_number_of_chunks( 64 * 1024 + 1 ) << std::endl;
    std::cout << get_number_of_chunks( 64 * 1024 - 1 ) << std::endl;
```


![image](https://user-images.githubusercontent.com/64067618/210072012-16459f0c-556e-42b1-b844-00083a38b176.png)

Reply:
Great line!